language/move-prover/bytecode-to-boogie/src/prelude.bpl

```diff
@@ -0,0 +1,543 @@
+// ================================================================================
+// Domains
+
+type TypeName;
+type FieldName = int;
+type LocalName;
+type Address = int;
+type ByteArray;
+type String;
+type {:datatype} Edge;
+function {:constructor} Field(f: FieldName): Edge;
```

Comment:
What does `{:constructor}` say to the solver?

Reply:
See above.